Block/Search.php

```diff
@@ -181,6 +201,30 @@ public function getMarkers()
                         'specialOpeningHours' => $retailer->getExtensionAttributes()->getSpecialOpeningHours(),
                     ]
                 );
+
+                    $promoList = $this->getPromoListByRetailerId($retailer->getId());
+                    $imageUrlPromotion = $this->getImageUrl().'/retailerpromotion/';
+                        foreach ($promoList as $promo) {
+                            $markerData['promotion'][] =
+                                [
+                                    'media'         => $imageUrlPromotion.$promo->getMediaPath(),
+                                    'title'         => $promo->getTitle(),
+                                    'description'   => $promo->getDescription(),
+                                ];
+                        }
+                    $servicesList = $this->getServiceListByRetailerId($retailer->getId());
+                    $imageUrlService = $this->getImageUrl().'/retailerservice/';
+                        foreach ($servicesList as $services) {
+
+                            $markerData['service'][] =
+                                [
+                                    'media'         => $imageUrlService.$services->getMediaPath(),
+                                    'title'         => $services->getName(),
+                                    'description'   => $services->getDescription(),
+                                ];
+                        }
+
+
```

Comment:
I'm not ok with this code. The module Storelocator must stay independant of the modules promotion and service.
You can use a plugin in each modules to add info in $markerData